### ISSUE TITLE
Sessions created with allow_multiple_sessions = on are not loaded after broker restart

### ISSUE THREAD

**Issue by Evaldas1984** [vernemq/vernemq#878]
### Environment

- VerneMQ Version: 1.5.0-1
- OS: Debian 9.5
- Erlang/OTP version (if building from source):

### Expected behavior

After broker restart- all sessions and offline messages are loaded from storage (level db).

### Actual behaviour

After broker restart- only sessions created when allow_multiple_sessions was off is loaded.

Before broker restart (p.s. aaa, bbb, ddd - sessions were create with allow_multiple_sessions = off, ccc- allow_multiple_sessions = on):
```
+--------------+----------------+-----+
|  client_id   |offline_messages|topic|
+--------------+----------------+-----+
|     bbbb     |       19       |test |
|     ddd      |       3        |test2|
|     aaa      |       37       |test |
|     ccc      |       3        |test |
|     ccc      |       3        |test2|
+--------------+----------------+-----+
After broker restart:
+--------------+----------------+-----+
|  client_id   |offline_messages|topic|
+--------------+----------------+-----+
|     bbbb     |       19       |test |
|     ddd      |       3        |test2|
|     aaa      |       37       |test |
+--------------+----------------+-----+
After connecting using client_id = ccc (no subscription command was issued, broker "remembered" sessions topics from before, but not offline messages)
+--------------+----------------+-----+
|  client_id   |offline_messages|topic|
+--------------+----------------+-----+
|     bbbb     |       19       |test |
|     ddd      |       0        |test2|
|     aaa      |       37       |test |
|     ccc      |       0        |test |
|     ccc      |       0        |test2|
+--------------+----------------+-----+
```

Steps to recreate problem:
1) In config file set allow_multiple_sessions = on.
2) restart broker
3) connect to broker with session id = ccc and clean session = false
4) subscribe to topic A  (QOS: 1)
5) disconnect
6) connect with publisher and send some messages to topic A (QOS: 1)
7) restart broker
8) connect to broker with session id = ccc and clean session = false
9) consumer won't receive any messages sent before connecting to broker at step 8

**Comment by larshesel** [comment]
Hi, thank you for the report and the detailed description. We'll look into it!

**Comment by dergraf** [comment]
The described behaviour is that way by design. A client that uses `allow_multiple_sessions = on` uses by design `clean_session = true`, actually VerneMQ doesn't even look at the provided `clean_session` setting. 
See: https://github.com/vernemq/vernemq/blob/master/apps/vmq_server/src/vmq_reg.erl#L130 (`_StartClean`)

`allow_multiple_sessions` has always been the black sheep among the non-standard MQTT options for us, as we haven't settled on a good way to make it work in a clustered setup, 
At the time we've introduced the Shared Subscription feature we saw many use cases for `allow_multiple_sessions` disappear, so instead of fixing/improving `allow_multiple_sessions` we highly encourage to look at Share Subscriptions. 

Have you looked at shared subscriptions? What is your use case with `allow_multiple_sessions`.
